main.py

```diff
@@ -80,42 +80,42 @@ def playing_11(request: Request):
             "squads": [squad1,squad2],
             "file": file,
             "match_type": match_type,
-            "teams": [request.query_params["team1"], request.query_params["team2"]],
+            "teams": [team1, team2],
         },
     )
 
 
 @app.post("/playing11")
-async def playing_11_post(request:Request,background_tasks: BackgroundTasks):
+async def playing_11_post(request:Request,file,type,team1,team2,background_tasks: BackgroundTasks):
     playing_11 = list(jsonable_encoder(await request.form()).keys())
     playing_11.remove("Confirm")
     players1 = '"' + '","'.join(playing_11[0:11]) + '"'
     players2 = '"' + '","'.join(playing_11[11:]) + '"'
     
     background_tasks.add_task(
         scrape_with_crochet,
-        file = request.query_params["file"],
-        match_type = request.query_params["type"],
-        team1 =request.query_params["team1"],
-        team2 = request.query_params["team2"],
+        file = file,
+        match_type = file,
+        team1 =file,
+        team2 = file,
         players2 = players2,
         players1 = players1
     )
     
     return RedirectResponse(
-        url = "/results?file="+request.query_params["file"], 
+        url = "/results?file="+file, 
         
     )
 
 
 @app.post("/results", response_class=HTMLResponse)
 def result(request: Request):
```

Comment:
Shouldnt file be included in the function definition header for results as well?